torchtext/experimental/vectors.py

```diff
@@ -1,7 +1,113 @@
 import csv
+import logging
+import os
+
 import torch
 from torch import Tensor
 import torch.nn as nn
+from tqdm import tqdm
+
+from torchtext.utils import (
+    download_from_url,
+    extract_archive
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _load_token_and_vectors_from_file(file_path):
+    stoi, tokens, vectors, dup_tokens = {}, [], [], []
+    with open(file_path, 'rb') as f:
+        for line in tqdm(f):
+            token, entries = line.rstrip().split(b" ", 1)
+            vector = torch.tensor([float(c) for c in entries.split(b" ")], dtype=torch.float)
+
+            try:
+                if isinstance(token, bytes):
+                    token = token.decode('utf-8')
+            except UnicodeDecodeError:
+                logger.info("Skipping non-UTF8 token {}".format(repr(token)))
```

Comment:
This should be an option. Also, don't we already have a unicode csv iterator?

Reply:
I think so but I wasn't sure if we could use this if we are loading in the file as a binary?